pkg/fuzzer: occasional TestFuzz failures

Once in a while, we observe failures like this (both on syzbot and on GitHub CI):

```
2025-01-14T12:36:25.6163042Z 2025/01/14 12:36:21 [FATAL] check failed: execution of simple program fails: call 0 failed with errno 998
```

From https://github.com/google/syzkaller/pull/5711#issuecomment-2621300420

```
--- FAIL: TestFuzz (11.22s)
    testutil.go:35: seed=0
Error: seed=0
    fuzzer_test.go:184: executor output:
Error: executor output:
        connected to manager: procs=4 cover_edges=1 kernel_64_bit=1 slowdown=1 syscall_timeout=50 program_timeout=5000 features=0x10
        received info reply: covfilter=0
        recv exec request 1: type=0 flags=0x3 env=0x1c09 exec=0x10 size=14
        proc 0: handshaking to execute request 1
        proc 0: got handshake reply
        proc 0: start executing request 1
        proc 0: got output: [9ms] exec opts: reqid=1 type=0 procid=0 threaded=1 cover=0 comps=0 dedup=0 signal=0  sandbox=1/0/0/0 timeouts=50/5000/1 kernel_64_bit=1
        proc 0: got output: spawned worker pid 90923
        proc 0: got output: Fatal glibc error: malloc assertion failure in sysmalloc: (old_top == initial_top (av) && old_size == 0) || ((unsigned long) (old_size) >= MINSIZE && prev_inuse (old_top) && ((unsigned long) old_end & (pagesize - 1)) == 0)
        proc 0: got output: Fatal glibc error: malloc assertion failure in sysmalloc: (old_top == initial_top (av) && old_size == 0) || ((unsigned long) (old_size) >= MINSIZE && prev_inuse (old_top) && ((unsigned long) old_end & (pagesize - 1)) == 0)
        proc 0: got execute reply
        handle completion: completed=0 output_size=262144
        proc 0: output pipe EOF
        proc 0: restarting subprocess, current state 2 attempts 0
        proc 0: subprocess exit status 6
        SYZFAIL: failed to recv rpc
        fd=3 want=4 recv=0 n=0 (errno 9: Bad file descriptor)
    fuzzer_test.go:185: executor process exited: exit status 67
```

Now the mystery is
1) Why do we corrupt the memory -- we don't execute any dangerous syscalls during the test.
2) Why and where we call malloc() from the executor process.

--- a/sys/targets/targets.go
+++ b/sys/targets/targets.go
@@ -702,7 +702,7 @@ func initTarget(target *Target, OS, arch string) {
 		target.NeedSyscallDefine = needSyscallDefine
 	}
 	if target.DataOffset == 0 {
-		target.DataOffset = 512 << 20
+		target.DataOffset = target.defaultDataOffset()
 	}
 	target.NumPages = (16 << 20) / target.PageSize
 	sourceDir := getSourceDir(target)
@@ -772,6 +772,15 @@ func initTarget(target *Target, OS, arch string) {
 		target.HostEndian = binary.BigEndian
 	}
 	target.initAddr2Line()
+}
+
+func (target *Target) defaultDataOffset() uint64 {
+	if target.PtrSize == 8 {
+		// An address from ASAN's 64-bit HighMem area.
+		return 0x200000000000
+	}
+	// From 32-bit HighMem area.
+	return 0x80000000
 }
 
 func (target *Target) initAddr2Line() {

